src/api/ethereum.ts

```diff
@@ -130,33 +129,19 @@ export const startFlow = async (
   });
   let call = [];
 
-  const isSubscribedUSDC = await idaContract.methods
-    .getSubscription(
-      USDCxAddress,
-      twoWayMarketAddress, // publisher
-      0, // indexId
-      sfUser.address,
-    )
-    .call();
+  // eslint-disable-next-line max-len
+  const config = indexIDA.filter((data) => data.input === inputTokenAddress && data.output === outputTokenAddress)[0];
 
-  const isSubscribedETH = await idaContract.methods
+  const isSubscribed = await idaContract.methods
     .getSubscription(
-      WETHxAddress,
+      config.output,
       twoWayMarketAddress, // publisher
```

Comment:
This variable is imported, use `exchangeAddress` which in this context will be the `twoWayMarketAddress`

Reply:
I have changed it here as well.